### PR TITLE
Test sign up

### PR DESCRIPTION
**Pull request by srwilliamg**
## Summary

- Added test for the endpoint /users
- Added validation of empty body
- Removed duplicate logic in errors file.

## Trello Card

https://trello.com/c/XQSSTnim/22-tests-de-sign-up

### CODE REVIEW

File: test/users.test.js [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+      .send(newUserData);
+
+    const {
+      status,
+      body: { name }
+    } = response;
+
+    expect(status).toBe(201);
+    expect(name).toBe('William');
+    done();
```
Comment (blueJayWays) [inline test/users.test.js]: is this `done();` necessary?

File: test/users.test.js [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+const truncateDatabase = require('./setup');
+const app = require('../app');
+
+beforeAll(async () => {
+  process.env.NODE_ENV = 'test';
+  await truncateDatabase();
+});
+
+describe('POST users', () => {
+  const newUserData = {
```
Comment (blueJayWays) [inline test/users.test.js]: It's a good idea to move this mocked json to another file inside test folder, and then import it.

File: test/users.test.js [hunk 3]
```diff
@@ -0,0 +1,93 @@
+const request = require('supertest');
+const truncateDatabase = require('./setup');
+const app = require('../app');
+
+beforeAll(async () => {
+  process.env.NODE_ENV = 'test';
```
Comment (blueJayWays) [inline test/users.test.js]: move `process.env.NODE_ENV` to the config file.

File: test/users.test.js [hunk 4]
```diff
@@ -0,0 +1,93 @@
...
+      .set('Accept', 'application/json')
+      .send(newUserData);
+
+    const {
+      status,
+      body: { message }
+    } = response;
+
+    expect(status).toBe(400);
+    expect(message).toBe('Email address already in use!');
```
Comment (blueJayWays) [inline test/users.test.js]: You should create a constant file, that contains all error messages. And then import it as error messages in the model and the tests.

File: test/users.test.js [hunk 5]
```diff
@@ -0,0 +1,93 @@
...
+
+describe('POST users', () => {
+  const newUserData = {
+    name: 'William',
+    lastName: 'Salazar',
+    email: 'william.salazar@wolox.co',
+    password: 'thisIsAVerySecurePassword'
+  };
+
+  test('create new user', async done => {
```
Comment (blueJayWays) [inline test/users.test.js]: you must agrupate singup test cases using [describe()](https://jestjs.io/docs/en/api#describename-fn). And name   test cases more specifically. for example "Should signup a new valid user".

File: test/users.test.js [hunk 6]
```diff
@@ -0,0 +1,93 @@
...
+      .set('Accept', 'application/json')
+      .send({ ...newUserData, password: 'min8' });
+
+    const {
+      status,
+      body: { message }
+    } = response;
+
+    expect(status).toBe(400);
+    expect(message).toBe('Validation error: Password length should be between 8 and 60');
```
Comment (blueJayWays) [inline test/users.test.js]: It's ideal to have only one expect per test case.

**Review by blueJayWays — CHANGES_REQUESTED** [review]

File: test/users.test.js (hunk 1 above)
Comment (srwilliamg) [reply test/users.test.js]: done, jest use it to know is the test works correctly or not.
I moved it to a Utils file where all the async errors are handle

File: test/users.test.js (hunk 2 above)
Comment (srwilliamg) [reply test/users.test.js]: Done :)

File: test/users.test.js (hunk 3 above)
Comment (srwilliamg) [reply test/users.test.js]: Done :)

File: test/users.test.js (hunk 4 above)
Comment (srwilliamg) [reply test/users.test.js]: Done

File: test/users.test.js (hunk 5 above)
Comment (srwilliamg) [reply test/users.test.js]: Done

File: test/users.test.js (hunk 6 above)
Comment (srwilliamg) [reply test/users.test.js]: Done

**Pull request merged**